### PR TITLE
Permissions: Reload list of sites, use default, confirm before any action, permissions.places-sites-limit (improvements), style clean up

### PR DESCRIPTION
**Pull request by janekptacijarabaci**
Ad https://github.com/MoonchildProductions/Pale-Moon/issues/823#issuecomment-294831241

> `about:permissions` does not allow to set the permission for site into "use default" state

Added a button.

> Once set using `about:permissions` the plugin state can't be cleared from the database the same way.

By "cleared" do you mean disabled? For global: I don't know how to do it better. See:
http://xref.palemoon.org/palemoon-trunk/search?string=.userDisabled&find=AddonManager.jsm

> `Forget About This Site` and `Remove [All] Cookies` buttons immediately remove the corresponding data without any warnings or confirmation dialogs.

It was added.

> `about:permissions` does not provide the filter to display only sites with modified permissions that could help to easily find the culprit in case of any problems.

Added configurable parameter (`about:config`): `permissions.places-sites-limit`.

> `about:permissions` does not allow to set the permission for site if target value equals current default.

> `pageinfo>permissions` has several mistakes in control logic which prevent to set specific permissions in case when global permissions for all sites differ from default values

__I don't have time to look at it just now...__

---

I've created the new build (x32, Windows) and tested.

### CODE REVIEW

**Comment by JustOff** [comment]
> Added a button.

In `pageinfo>permissions` set permission for site into "use default" state means clear corresponding record from `permissions.sqlite`, not set it to default value as you made.

> By "cleared" do you mean disabled?

See above.

> Added configurable parameter (`about:config`): `permissions.places-sites-limit`.

You mean set it to 0 for view only sites with modified permissions? Then this trick works.

PS: If I would have decided to work on `about:permissions` task I probably just use the view and code for single site from `pageinfo>permissions` as it has almost all necessary (including default state handling) and works almost fine. But since I'm not, I can only advise to do so.

**Comment by janekptacijarabaci** [comment]
Thank you for the feedback.

> You mean set it to 0 for view only sites with modified permissions?

Yes.

File: browser/locales/en-US/chrome/browser/preferences/aboutPermissions.properties [hunk 1]
```diff
@@ -2,6 +2,10 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+# LOCALIZATION NOTE (permissionsConfirm): Confirm before any action.
+# See: http://developer.mozilla.org/en/docs/Localization_and_Plurals
+permissionsConfirm=Are you sure you want to continue?
```
Comment (wolfbeast) [inline browser/locales/en-US/chrome/browser/preferences/aboutPermissions.properties]: Very/too vague... continue with what?

File: browser/locales/en-US/chrome/browser/preferences/aboutPermissions.dtd [hunk 2]
```diff
@@ -15,8 +15,12 @@
 
 <!ENTITY header.defaults                 "Default Permissions for All Sites">
 
+<!ENTITY permissions.sitesReload         "Reload list of sites">
+
 <!ENTITY permissions.forgetSite          "Forget About This Site">
 
+<!ENTITY permission.default              "Set a default value">
```
Comment (wolfbeast) [inline browser/locales/en-US/chrome/browser/preferences/aboutPermissions.dtd]: Set a default value for what? What is the default value? Very vague...

If this is the button label, IIUC, then it should just be "Use Default" or similar.

File: browser/components/preferences/aboutPermissions.js [hunk 3]
```diff
@@ -2,44 +2,52 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-let Ci = Components.interfaces;
-let Cc = Components.classes;
-let Cu = Components.utils;
+var Cc = Components.classes;
+var Ci = Components.interfaces;
+var Cu = Components.utils;
```
Comment (wolfbeast) [inline browser/components/preferences/aboutPermissions.js]: Why are you changing all these `let` keywords to `var`??

File: browser/components/preferences/aboutPermissions.js [hunk 4]
```diff
@@ -117,7 +125,7 @@ Site.prototype = {
    *
    * @param aType
    *        The permission type string stored in permission manager.
-   *        e.g. "cookie", "geo", "indexedDB", "popup", "image"
+   *        e.g. "image", "popup", "cookie", "geo", "indexedDB"
```
Comment (wolfbeast) [inline browser/components/preferences/aboutPermissions.js]: Why are you changing the order here?

File: browser/components/preferences/aboutPermissions.js [hunk 5]
```diff
@@ -156,7 +164,7 @@ Site.prototype = {
...
```
Comment (wolfbeast) [inline browser/components/preferences/aboutPermissions.js]: Ditto, for order?

File: browser/components/preferences/aboutPermissions.js [hunk 6]
```diff
@@ -185,7 +193,7 @@ Site.prototype = {
...
```
Comment (wolfbeast) [inline browser/components/preferences/aboutPermissions.js]: ditto, for order?

File: browser/components/preferences/aboutPermissions.js [hunk 7]
```diff
@@ -95,7 +103,7 @@ Site.prototype = {
     let rev_host = this.host.split("").reverse().join("") + ".";
     gVisitStmt.params.rev_host = rev_host;
     gVisitStmt.executeAsync({
-      handleResult: function(aResults) {
+      handleResult: function (aResults) {
```
Comment (wolfbeast) [inline browser/components/preferences/aboutPermissions.js]: Why are you adding spaces between function and parentheses...?
common coding style is `function()` not `function ()` -- please don't make unnecessary and counter-practice style changes.

(also for all others below where you did this)

File: browser/components/preferences/aboutPermissions.js [hunk 8]
```diff
@@ -646,36 +685,40 @@ let AboutPermissions = {
   },
 
   observe: function (aSubject, aTopic, aData) {
-    switch(aTopic) {
+    switch (aTopic) {
```
Comment (wolfbeast) [inline browser/components/preferences/aboutPermissions.js]: Same issue as with `function()` -- common style = no space, please don't change this.

**Comment by JustOff** [comment]
Btw, are you intentionally use different font style for 'Use default' buttons?

palemoon_2017-04-19_22-27-14

**Comment by janekptacijarabaci** [comment]
@JustOff 
> are you intentionally use different font style for 'Use default' buttons?

It's only the smaller font. Do you have a better idea? Maybe another font style?! I don't know...

**Comment by JustOff** [comment]
> I don't know...

Ok, never mind.

**Comment by wolfbeast** [comment]
There's still the question "Are you sure you want to continue?" -- when is this used? It's too vague.

**Comment by wolfbeast** [comment]
Please remove the confirmation dialog boxes. It's unnecessary and just annoying.

**Comment by janekptacijarabaci** [comment]
@wolfbeast 
Ad "aboutPermisions.xul and (thrice):
https://github.com/MoonchildProductions/Pale-Moon/pull/1043/files#diff-9b3cb3249f3a1f077eac471d10b882baR64

Please replace it for what you think is best.

**Comment by JustOff** [comment]
> Please remove the confirmation dialog boxes. It's unnecessary and just annoying.

Sorry for insisting, but I think it's very dangerous decision. We probably may just remove 'Remove Cookies' buttons at all 'cause such action can be done using 'Manage Cookies' either one by one or completely. And as for 'Forget About This Site' button I'm sure the confirmation dialog should be there.

**Comment by janekptacijarabaci** [comment]
> Please remove the confirmation dialog boxes.

Unfortunately, I can't do it now.

**Comment by wolfbeast** [comment]
@JustOff we don't ask for permissions in other places where you have these options, there is no reason whatsoever to suddenly introduce it here, which is supposed to be an interface for quick management of site permissions and per-site settings.

None of these settings are dangerous to use.

@janekptacijarabaci All right, I'll merge it and remove it myself.

**Comment by wolfbeast** [comment]
@JustOff FYI: this is going to be the last accepted commit with locale changes for 27.3.

**Comment by mattatobin** [comment]
If they can't in time it would be best to make sure untraslated english strings are at least existant for all active langpacks by release day.

**Comment by JustOff** [comment]
@mattatobin 
That will be done automatically next time I build the project on CrowdIn.

**Pull request merged**